x/axelarnet/keeper/msg_server.go

```diff
@@ -79,7 +80,9 @@ func (s msgServer) CallContract(c context.Context, req *types.CallContractReques
 	// axelar gateway expects keccak256 hashes for payloads
 	payloadHash := crypto.Keccak256(req.Payload)
 
-	msg := nexus.NewGeneralMessage(s.nexus.GenerateMessageID(ctx), sender, recipient, payloadHash, nexus.Sent, nil)
+	txID := sha256.Sum256(ctx.TxBytes())
```

Comment:
oh, this is also duplicated. Now I'm thinking that we should one of the following:
- return tx ID from GenerateMessageID too
- Split GenerateMessageID into two methods, once that give txID, nonce, and another that creates message ID from it
- Change NewGeneralMessage to take in a txID and txIndex, and it'll format it appropriately. GenerateMesageID changes to GenerateMessageNonce which returns the nonce/txIndex

The last option is the cleanest to me, it also works for EVM -> Cosmos

Reply:
Based on the conversation [here](https://github.com/axelarnetwork/axelar-core/pull/1907#discussion_r1152125008), I'm ok with refactoring this later